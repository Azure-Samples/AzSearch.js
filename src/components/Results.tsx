import * as React from "react";
import { PropsType } from "../containers/ResultsContainer";
import * as objAssign from "object-assign";
import { defaultCss } from "../utils/css";

export type State = {};

class Results extends React.PureComponent<PropsType, State> {
    render() {
        const { results, template, skip, top, count } = this.props;
        let css = objAssign({}, defaultCss, this.props.css);
        let countElement = count > 0 ? ` of ${count}` : "";
        let bottomRange = skip + 1;
        let topRange = skip + top;
        topRange = topRange > count ? count : topRange;
        let resultsBlurb =
            <div className={css.results__blurb}>
                {bottomRange} - {topRange}{countElement}
            </div>;
        resultsBlurb = results.length > 0 ? resultsBlurb : <div></div>;

        const renderedResults = results.map((result: any, index: number) => {
            let html = template ? template.render(result) : null;
            if (html) {
                return (
                    <div className={css.searchResults__result} key={index} dangerouslySetInnerHTML={{ __html: html }} />
                );
            }
            else {
                return (
                    <div className={css.searchResults__result} key={index} >
                        <pre>
                            <code>
                                {JSON.stringify(result, null, 4)}
                            </code>
                        </pre>
                    </div>
                );
            }
        });

        return (
            <div>
                {resultsBlurb}
                {renderedResults}
            </div>
        );
    }
}

export default Results;