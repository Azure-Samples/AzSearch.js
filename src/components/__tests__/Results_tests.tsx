/// <reference path="../../../dist/hogan.js.d.ts" />
import * as React from "react";
import Results from "../Results";
import { shallow } from "enzyme";
import { compile } from "hogan.js";

let results = [
    {
        key: "foo",
        text: "bar"
    },
    {
        key: "fizz",
        text: "buzz"
    }
];
const template = compile("{{key}}");

describe("<Results/>", () => {
    it("renders dummy snapshot", () => {
        expect(shallow(
            <Results template={null} css={null} results={results} top={50} skip={0} count={results.length} />
        )).toMatchSnapshot();
    });
    it("renders two results", () => {
        const wrapper = shallow(
            <Results template={null} css={null} results={results} top={50} skip={0} count={results.length} />
        );
        expect(wrapper.find(".searchResults__result").length).toBe(2);
        expect(wrapper.find(".results__blurb").text()).toEqual("1 - 2 of 2");
    });
    it("renders empty div for zero results", () => {
        const wrapper = shallow(
            <Results template={null} css={null} results={[]} top={50} skip={0} count={results.length} />
        );
        expect(wrapper.find(".searchResults__result").length).toBe(0);
        expect(wrapper.find("div").length).toBe(2);
    });
    it("renders two results with custom template", () => {
        const wrapper = shallow(
            <Results template={template} css={null} results={results} top={50} skip={0} count={results.length} />
        );
        const resultsElms = wrapper.find(".searchResults__result");
        expect(resultsElms.length).toBe(2);
        expect(resultsElms.first().html()).toEqual("<div class=\"searchResults__result col-xs-12 col-sm-12\">foo</div>");
        expect(wrapper.find(".results__blurb").text()).toEqual("1 - 2 of 2");
    });
});