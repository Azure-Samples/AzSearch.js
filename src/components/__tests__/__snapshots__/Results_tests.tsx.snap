// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results/> renders dummy snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="results__blurb row"
    >
        1
         - 
        2
         of 2
    </div>
    <div
        className="searchResults__result col-xs-12 col-sm-12"
    >
        <pre>
            <code>
                {
                    "key": "foo",
                    "text": "bar"
                }
            </code>
        </pre>
    </div>
    <div
        className="searchResults__result col-xs-12 col-sm-12"
    >
        <pre>
            <code>
                {
                    "key": "fizz",
                    "text": "buzz"
                }
            </code>
        </pre>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="results__blurb row"
      >
            1
             - 
            2
             of 2
      </div>
      <div
            className="searchResults__result col-xs-12 col-sm-12"
      >
            <pre>
                  <code>
                        {
                            "key": "foo",
                            "text": "bar"
                        }
                  </code>
            </pre>
      </div>
      <div
            className="searchResults__result col-xs-12 col-sm-12"
      >
            <pre>
                  <code>
                        {
                            "key": "fizz",
                            "text": "buzz"
                        }
                  </code>
            </pre>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Results
        count={2}
        css={null}
        results={
                Array [
                        Object {
                          "key": "foo",
                          "text": "bar",
                        },
                        Object {
                          "key": "fizz",
                          "text": "buzz",
                        },
                      ]
        }
        skip={0}
        template={null}
        top={50}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Results {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "count": 2,
          "css": null,
          "results": Array [
            Object {
              "key": "foo",
              "text": "bar",
            },
            Object {
              "key": "fizz",
              "text": "buzz",
            },
          ],
          "skip": 0,
          "template": null,
          "top": 50,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="results__blurb row"
          >
                    1
                     - 
                    2
                     of 2
          </div>
          <div
                    className="searchResults__result col-xs-12 col-sm-12"
          >
                    <pre>
                              <code>
                                        {
                                            "key": "foo",
                                            "text": "bar"
                                        }
                              </code>
                    </pre>
          </div>
          <div
                    className="searchResults__result col-xs-12 col-sm-12"
          >
                    <pre>
                              <code>
                                        {
                                            "key": "fizz",
                                            "text": "buzz"
                                        }
                              </code>
                    </pre>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="results__blurb row"
          >
                    1
                     - 
                    2
                     of 2
          </div>
          <div
                    className="searchResults__result col-xs-12 col-sm-12"
          >
                    <pre>
                              <code>
                                        {
                                            "key": "foo",
                                            "text": "bar"
                                        }
                              </code>
                    </pre>
          </div>
          <div
                    className="searchResults__result col-xs-12 col-sm-12"
          >
                    <pre>
                              <code>
                                        {
                                            "key": "fizz",
                                            "text": "buzz"
                                        }
                              </code>
                    </pre>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Results
    count={2}
    css={null}
    results={
        Array [
            Object {
              "key": "foo",
              "text": "bar",
            },
            Object {
              "key": "fizz",
              "text": "buzz",
            },
          ]
    }
    skip={0}
    template={null}
    top={50}
/>,
}
`;
